traffic_ops/traffic_ops_golang/federation_resolvers/federation_resolvers.go

```diff
@@ -162,19 +176,31 @@ func Read(w http.ResponseWriter, r *http.Request) {
 	}
 	defer rows.Close()
 
-	var resolvers = []tc.FederationResolver{}
 	for rows.Next() {
-		var resolver tc.FederationResolver
-		if err := rows.Scan(&resolver.ID, &resolver.IPAddress, &resolver.LastUpdated, &resolver.Type); err != nil {
-			userErr, sysErr, errCode = api.ParseDBError(err)
-			if sysErr != nil {
-				sysErr = fmt.Errorf("federation_resolver scanning: %v", sysErr)
+		// Based on version we load types - for version 5 and above we use FederationResolverV5
+		if inf.Version != nil && inf.Version.Major >= 5 && inf.Version.Minor >= 0 {
+			var resolver tc.FederationResolverV5
+			if err := rows.Scan(&resolver.ID, &resolver.IPAddress, &resolver.LastUpdated, &resolver.Type); err != nil {
+				userErr, sysErr, errCode = api.ParseDBError(err)
+				if sysErr != nil {
+					sysErr = fmt.Errorf("federation_resolver scanning: %v", sysErr)
+				}
+				api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+				return
 			}
-			api.HandleErr(w, r, tx, errCode, userErr, sysErr)
-			return
+			resolvers = append(resolvers, resolver)
```

Comment:
Instead of repeating this block twice, here's what I'd do:
Scan the struct into a v4 struct as it was being done earlier. Then, check for the version, and if it's >= 5.0, then convert the v4 struct into a v5 struct before appending it to the interface array `resolvers`.